src/System.Runtime/tests/System/String.cs

```diff
@@ -94,7 +94,7 @@ public static void TestCtorCharIntInvalid()
     [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h' }, 0, 8, "abcdefgh")]
     [InlineData(new char[] { 'a', 'b', 'c', 'd', '\0', 'e', 'f', 'g', 'h' }, 0, 9, "abcd\0efgh")]
     [InlineData(new char[] { 'П', 'Р', 'И', 'В', 'Е', 'Т' }, 0, 6, "ПРИВЕТ")]
```

Comment:
The use of `\0` here seem to be concerned about correct behaviour on the null char. Maybe it would be worth adding other troublesome characters like other controls or non-characters like U+FFFF?